### PR TITLE
Polls: show warning about undecryptable relations

### PR DESCRIPTION
**Pull request by kerryarchibald**
With: https://github.com/matrix-org/matrix-js-sdk/pull/3163

Show a warning message when a poll has undecryptable relations so users know their vote count may be inaccurate
Screenshot 2023-02-17 at 11 38 23

## Checklist

-   [ ] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Polls: show warning about undecryptable relations ([\#10179](https://github.com/matrix-org/matrix-react-sdk/pull/10179)). Contributed by @kerryarchibald.

### CODE REVIEW

File: src/utils/poll/countPollDecryptionFailures.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Relations } from "matrix-js-sdk/src/models/relations";
+
+export const countPollDecryptionFailures = (responseRelations: Relations): number =>
```
Comment (weeman1337) [inline src/utils/poll/countPollDecryptionFailures.ts:19]: I cannot find any usage of this in the PR.
Does a poll not already provide the number [here](https://github.com/matrix-org/matrix-js-sdk/pull/3163/files#diff-6d5fabe09478e35a518f33e36f448a7e7c3e224ffa9a4a726fe8ccfeefd7b4f0R101)?

**Review by weeman1337 — CHANGES_REQUESTED** [review]
Some unused code may have crept in. Otherwise it looks good.

File: src/components/views/messages/MPollBody.tsx [hunk 2]
```diff
@@ -182,12 +182,14 @@ export default class MPollBody extends React.Component<IBodyProps, IState> {
     private addListeners(): void {
         this.state.poll?.on(PollEvent.Responses, this.onResponsesChange);
         this.state.poll?.on(PollEvent.End, this.onRelationsChange);
+        this.state.poll?.on(PollEvent.UndecryptableRelations, this.render.bind(this));
```
Comment (weeman1337) [inline src/components/views/messages/MPollBody.tsx:185]: Instead of calling render, would it be better to add the undecryptable event count to the state?

Comment (kerryarchibald) [reply src/components/views/messages/MPollBody.tsx:185]: I agree that this is kind of weird, but I also can't think of a concrete reason that duplicating the event count in MPollBody state would be better.

**Review by weeman1337 — APPROVED** [review]

**Pull request merged**